### PR TITLE
Bug 2090842: Improve event filtering with proper type check

### PR DESCRIPTION
**Pull request by jc-rh**
The fix is all in the `SetupWithManager` function in the controller. The rest of the change is just cleanup from previous changes.

### CODE REVIEW

**Review by sudomakeinstall2 — APPROVED** [review]

File: controllers/clustergroupupgrade_controller.go [hunk 1]
```diff
@@ -1885,32 +1886,30 @@ func (r *ClusterGroupUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error
...
 			},
-			CreateFunc: func(ce event.CreateEvent) bool {
-				// only interested in CGU create event
-				if ce.Object.GetLabels()["app.kubernetes.io/instance"] == "policies" {
-					return false
-				}
-				return true
+			CreateFunc:  func(ce event.CreateEvent) bool { return true },
+			GenericFunc: func(ge event.GenericEvent) bool { return false },
+			DeleteFunc:  func(de event.DeleteEvent) bool { return true },
```
[the remark this answers is not in the document]
Comment (jc-rh) [reply controllers/clustergroupupgrade_controller.go:1900]:
```
// Delete implements Predicate.
func (p Funcs) Delete(e event.DeleteEvent) bool {
	if p.DeleteFunc != nil {
		return p.DeleteFunc(e)
	}
	return true
}
```
The default seems to be true if function is not implemented. For policy, it's causing repeated reconcile calls. I didn't take care of this part initially as these calls were light and they don't impact our scale test rate significantly. Now I am following your steps to have a complete filter implementation. Good point on update event for deletion. I should flip the cgu delete to false then.

[the remark this answers is not in the document]
Comment (jc-rh) [reply controllers/clustergroupupgrade_controller.go]: Updated

**Review by Missxiaoguo — APPROVED** [review]
Looks good

**Pull request merged**